src/main/java/io/supertokens/inmemorydb/Start.java

```diff
@@ -1619,4 +1627,86 @@ public void addInfoToNonAuthRecipesBasedOnUserId(String className, String userId
             throw new IllegalStateException("ClassName: " + className + " is not part of NonAuthRecipeStorage");
         }
     }
+
+    // TOTP recipe: 
+
+    @Override
+    public void createDevice(TOTPDevice device) throws StorageQueryException {
```

Comment:
shouldn't this throw a `DeviceAlreadyExistsException`? This would happen if the sql query fails with a primary key already exists error. See how other queries are done in this PR

Reply:
Done.